ext/InfiniteDisjunctiveProgramming.jl

```diff
@@ -0,0 +1,181 @@
+module InfiniteDisjunctiveProgramming
+
+import JuMP.MOI as _MOI
+import InfiniteOpt, JuMP
+import DisjunctiveProgramming as DP
+
+################################################################################
+#                                   MODEL
+################################################################################
+function DP.InfiniteGDPModel(args...; kwargs...)
+    return DP.GDPModel{
+        InfiniteOpt.InfiniteModel, 
+        InfiniteOpt.GeneralVariableRef, 
+        InfiniteOpt.InfOptConstraintRef
+        }(args...; kwargs...)
+end
+
+function DP.collect_all_vars(model::InfiniteOpt.InfiniteModel)
+    vars = collect(JuMP.all_variables(model))
+    derivs = collect(InfiniteOpt.all_derivatives(model))
```

Comment:
Here, `collect` is unnecessary, the functions already provide you with a vector.